WordPress/gutenberg
# Block Support: Add height block support feature

Related: https://github.com/WordPress/gutenberg/issues/28356
Depends on: https://github.com/WordPress/gutenberg/pull/32392

## Description
Adds height block support and includes in new Dimensions panel.

## How has this been tested?
Theme.json test:
`wp-env run phpunit 'phpunit -c /var/www/html/wp-content/plugins/gutenberg/phpunit.xml.dist --verbose --filter stylesheet /var/www/html/wp-content/plugins/gutenberg/phpunit/class-wp-theme-json-test.php'`

Style.js test:
`npm run test-unit:watch packages/block-editor/src/hooks/test/style.js`

#### Manual Test Instructions

1. Checkout this PR and build
2. Enable custom dimensions support including height, padding and margin - [theme.json gist](https://gist.github.com/aaronrobertshaw/5556efc230d25f337d9ffe5ff9580e2d)
3. Enable dimensions block support for Group block - [block.json gist](https://gist.github.com/aaronrobertshaw/3532aa40d7b154ffdef720e7d249705a)
4. Create a post, add a group block and select it
5. Ensure dimensions panel appears in the sidebar
6. Open the dimension panel's more menu and toggle on the height control
7. Adjust the height can confirm the block's height gets adjusted. Inspect element in dev tools to confirm CSS rule
8. Save post and view on the frontend. The height style should be applied correctly.
9. Repeat the process for a dynamic block, ensuring correct styles on the frontend.
10. Switch to the site editor, open the global styles sidebar and select the `By Block Type` tab.
11. Confirm the global styles dimensions panel includes height for the group block and that it works.

## Screenshots 
<img src="https://user-images.githubusercontent.com/60436221/121115463-74f7b480-c858-11eb-94a8-2d7db9ed0da8.gif" width="500"/>

## Types of changes
New feature.

## Next Steps

- [x] Possibly add min/max height support alongside this? 
PR for min-height: https://github.com/WordPress/gutenberg/pull/33970

## Checklist:
